### PR TITLE
adds captain sword

### PR DESCRIPTION
**Pull request by ParadoxSpace**
Adds the Captain sabre, among other things. The sabre in question requires the sheath to be in your inventory, is basically just a regular sword otherwise. Other swords could be made to use sheathes later. This has been added in a sword rack to the Captain's office. The ATM was made tile bound only so that it doesn't phase through walls. 

I put the antique laser pistol back.

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
image
image
image

**Comment by veraknocks** [comment]
i prefer the, in my experience, rarely abused captains laser. as cool as samurai and sword memes are the gun is more practical for those who choose to acquire it as opposed to the mere asethetic value of the Royal Stick

**Comment by ParadoxSpace** [comment]
!review

**Comment by Sindorman** [comment]
I would agree with veraknocks. Laser should remain, it is an iconic item. This might need a feedback thread. Please make it.

**Comment by ParadoxSpace** [comment]
Iconic =/= good.

**Comment by ParadoxSpace** [comment]
Since the forums take 45 minutes each page to load, can I just map the laser back in instead?

**Comment by ParadoxSpace** [comment]
Laspistol has been returned, adjust your thumbs accordingly.

**Comment by LordFowl** [comment]
Giving the captain both a laser and a sword is a little much. This might need a feedback thread. Please make it.

**Comment by ParadoxSpace** [comment]
It took me like a combined fourteen minutes but here. https://forums.aurorastation.org/topic/11788-captain-sabrelaser-thread/

File: code/game/objects/items/weapons/material/swords.dm [hunk 1]
```diff
@@ -147,4 +147,28 @@
...
+	item_state = "redsabre"
+	contained_sprite = 1
+	unbreakable = 1
+	attack_verb = list("slashed", "cut", "chopped", "torn")
+	applies_material_colour = 0
+	slot_flags = 0
+
+/obj/item/weapon/material/sword/captain_sword/on_exit_storage(obj/item/weapon/storage/S)
+	..()
+	var/obj/item/weapon/storage/belt/sabre/B = S
```
Comment (skull132) [inline code/game/objects/items/weapons/material/swords.dm]: Typecast argument as a saber, you don't need another variable for this.

File: code/game/objects/items/weapons/material/swords.dm [hunk 2]
```diff
@@ -147,4 +147,28 @@
...
+	if(istype(B))
+		playsound(B, 'sound/items/unsheath.ogg', 25, 1)
+
+/obj/item/weapon/material/sword/captain_sword/on_enter_storage(obj/item/weapon/storage/S)
+	..()
+	var/obj/item/weapon/storage/belt/sabre/B = S
```
Comment (skull132) [inline code/game/objects/items/weapons/material/swords.dm]: Same here.

File: code/game/objects/items/weapons/storage/belt.dm [hunk 3]
```diff
@@ -372,4 +372,46 @@
...
+	w_class = 4
+	max_w_class = 4
+	attack_verb = list("struck", "bashed", "disciplined")
+	use_sound = null
+	can_hold = list(
+		/obj/item/weapon/material/sword/captain_sword
+		)
+
+/obj/item/weapon/storage/belt/sabre/AltClick(mob/user)
+	if(!usr || usr.stat || usr.lying || usr.restrained() || !Adjacent(usr))
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/belt.dm]: Use of `usr` in a non-interface proc. No, refactor to user. And use `use_check(user)`.

File: code/game/objects/items/weapons/storage/belt.dm [hunk 4]
```diff
@@ -372,4 +372,46 @@
...
+	else
+		to_chat(user, "[src] is empty.")
+
+/obj/item/weapon/storage/belt/sabre/update_icon()
+	icon_state = "sheath"
+	item_state = "sheath"
+	if(contents.len)
+		icon_state += "-redsabre"
+		item_state += "-redsabre"
+	if(loc && istype(loc, /mob/living))
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/belt.dm]:
```suggestion
	if(istype(loc, /mob/living))
```

**Comment by ParadoxSpace** [comment]
that should do it chief

**Comment by skull132** [comment]
Giving the captain a sword does not well suit our setting. Swords are generally associated with the Unathi and ninja in our setting, and definitely not with a common Biesel human.

**Comment by ParadoxSpace** [comment]
This day extracts a heavy toll.

**Pull request closed without merging**